app/actions/remote/channel.ts

```diff
@@ -1107,6 +1108,12 @@ export async function switchToChannelById(serverUrl: string, channelId: string,
     markChannelAsRead(serverUrl, channelId);
     fetchChannelStats(serverUrl, channelId);
 
+    getChannelById(database, channelId).then((channel) => {
+        if (channel?.isGroupConstrained) {
+            fetchGroupsForChannel(serverUrl, channelId);
```

Comment:
The check about constraint or not can happen inside the action in itself

Reply:
Yup updated to handle like I do for team switch 👍